### PR TITLE
bug:1691546 fix(catalog): do not add owner references to clusterroles or crbs

### PR DESCRIPTION
**Pull request by ecordell**
Operators requesting Cluster Permissions could hit a kube GC bug

### CODE REVIEW

File: pkg/controller/operators/catalog/operator.go [hunk 1]
```diff
@@ -1168,16 +1168,11 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
...
-				if err != nil {
-					return errorwrap.Wrapf(err, "error generating ownerrefs for clusterrole %s", cr.GetName())
-				}
-				cr.OwnerReferences = updated
-
 				// Attempt to create the ClusterRole.
 				_, err = o.OpClient.KubernetesInterface().RbacV1().ClusterRoles().Create(&cr)
 				if k8serrors.IsAlreadyExists(err) {
+					// if we're updating, point owner to the newest csv
+					cr.Annotations[ownerutil.OwnerKey] = step.Resolving
```
Comment (njhale) [inline pkg/controller/operators/catalog/operator.go]: Isn't this supposed to be a label?

Comment (ecordell) [reply pkg/controller/operators/catalog/operator.go]: It should be a label, yes. 

We're not currently GCing at all, so switching it should be fine as well. We do need to have an api similar to ownerrefs though, and GC these ourselves (or implement GC in kube)

File: pkg/controller/operators/catalog/operator.go [hunk 2]
```diff
@@ -1198,17 +1193,11 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
...
-				updated, err := o.getUpdatedOwnerReferences(rb.OwnerReferences, plan.Namespace)
-				if err != nil {
-					return errorwrap.Wrapf(err, "error generating ownerrefs for clusterrolebinding %s", rb.GetName())
-				}
-				rb.OwnerReferences = updated
-
 				// Attempt to create the ClusterRoleBinding.
 				_, err = o.OpClient.KubernetesInterface().RbacV1().ClusterRoleBindings().Create(&rb)
 				if k8serrors.IsAlreadyExists(err) {
-					rb.SetNamespace(plan.Namespace)
```
Comment (jpeeler) [inline pkg/controller/operators/catalog/operator.go]: Why was this line removed?

Comment (ecordell) [reply pkg/controller/operators/catalog/operator.go]: ClusterRoleBindings don't have namespaces :)

Comment (jpeeler) [reply pkg/controller/operators/catalog/operator.go]: Ah right!

File: pkg/controller/operators/catalog/operator.go [hunk 3]
```diff
@@ -1168,16 +1168,11 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
...
-				if err != nil {
-					return errorwrap.Wrapf(err, "error generating ownerrefs for clusterrole %s", cr.GetName())
-				}
-				cr.OwnerReferences = updated
-
 				// Attempt to create the ClusterRole.
 				_, err = o.OpClient.KubernetesInterface().RbacV1().ClusterRoles().Create(&cr)
 				if k8serrors.IsAlreadyExists(err) {
+					// if we're updating, point owner to the newest csv
+					cr.Labels[ownerutil.OwnerKey] = step.Resolving
```
Comment (tkashem) [inline pkg/controller/operators/catalog/operator.go]: question: how is this label `ownerutil.OwnerKey` used?

Comment (ecordell) [reply pkg/controller/operators/catalog/operator.go]: it's a label that points to the owner for cases where we can't use real ownerrefs
